### ISSUE TITLE
Generator not compatible with "type": "module".

### ISSUE THREAD

**Issue by MrNossiom** [ivanhofer/typesafe-i18n#80]
Hello Ivan,

I switched to full ESM modules and everything works except the generated file `i18n-util.ts`.

The import statement isn't compatible with ESM, since it imports without the `.js` extension and tries to import a folder directly.

Maybe there is a way to add support via a flag. (like `esmSupport`)

Something like this at https://github.com/ivanhofer/typesafe-i18n/blob/main/packages/generator/src/files/generate-util.ts#L114 
```diff
- import { initFormatters } from './${formattersTemplatePath}'
+ import { initFormatters } from './${formattersTemplatePath}${esmSupport ? '.js' : ''}'
```

I can make a pull request if you accept with all necessary types and code changes.

I'm at your disposal for any questions

MrNossiom
Thanks

**Comment by MrNossiom** [comment]
This might be a duplicate of #71, but I wanted to create my own issue.

**Comment by ivanhofer** [comment]
Hi @MrNossiom,
sounds like a good solution! You can create a PR if you like.
I'm not so sure about #71, but we will see once the imports are updated

**Comment by ivanhofer** [comment]
Fixed in version `2.39.0`
